sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -749,6 +790,48 @@ public void testSinkWithSendErrors() throws Throwable {
   }
 
   @Test
+  public void testUnboundedSourceStartReadTime() {
+
+    if (new ConsumerSpEL().hasOffsetsForTimes()) {
+      int numElements = 1000;
+      int startTime = numElements / 20 / 2; // half elements
```

Comment:
In our MockConsumer, we let the elements of the time and offset equal and there are 20 partitions. So set this startTime can read half elements.